clippy_lints/src/methods/mod.rs

```diff
@@ -2677,10 +2677,10 @@ fn lint_filter_flat_map<'a, 'tcx>(
 ) {
     // lint if caller of `.filter().flat_map()` is an Iterator
     if match_trait_method(cx, expr, &paths::ITERATOR) {
-        let msg = "called `filter(p).flat_map(q)` on an `Iterator`. \
-                   This is more succinctly expressed by calling `.flat_map(..)` \
-                   and filtering by returning an empty Iterator.";
-        span_lint(cx, FILTER_MAP, expr.span, msg);
+        let msg = "called `filter(p).flat_map(q)` on an `Iterator`";
+        let hint = "This is more succinctly expressed by calling `.flat_map(..)` \
+            and filtering by returning an empty Iterator.";
```

Comment:
```suggestion
        let hint = "this is more succinctly expressed by calling `.flat_map(..)` \
            and filtering by returning `iter::empty()`";
```

Let's suggest [`std::iter::empty`](https://doc.rust-lang.org/std/iter/fn.empty.html) here.